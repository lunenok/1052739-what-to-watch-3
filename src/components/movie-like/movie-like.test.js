import React from "react";
import renderer from "react-test-renderer";
import MovieLike from "./movie-like.jsx";

const movieTestList = [
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    genre: `Drama`,
    year: 2019,
    imgSrc: `war-of-the-worlds.jpg`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 2,
    title: `Battlefield Earth: A Saga of the Year 3000`,
    genre: `Drama`,
    year: 2019,
    imgSrc: `war-of-the-worlds.jpg`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }
];

it(`Should movie like-list render correctly`, () => {
  const tree = renderer
    .create(
        <MovieLike
          movieLikeThis={movieTestList}
          onMovieClick={()=>{}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
