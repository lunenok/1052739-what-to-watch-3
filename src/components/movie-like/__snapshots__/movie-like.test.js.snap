// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should movie like-list render correctly 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  >
    <a
      aria-current={null}
      className="small-movie-card catalog__movies-card"
      href="/movie/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "#c9b37e",
        }
      }
    >
      <article
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        />
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
    </a>
  </div>
</section>
`;
